server/src/main/java/com/thoughtworks/go/server/service/ElasticAgentPluginService.java

```diff
@@ -311,9 +330,25 @@ public void jobCompleted(JobInstance job) {
 
         ElasticProfile elasticProfile = job.getPlan().getElasticProfile();
         ClusterProfile clusterProfile = job.getPlan().getClusterProfile();
-        Map<String, String> elasticProfileConfiguration = elasticProfile.getConfigurationAsMap(true);
-        Map<String, String> clusterProfileConfiguration = clusterProfile != null ? clusterProfile.getConfigurationAsMap(true) : Collections.EMPTY_MAP;
-
+        secretParamResolver.resolve(elasticProfile);
+        Map<String, String> elasticProfileConfiguration = elasticProfile.getConfigurationAsMap(true, true);
+        Map<String, String> clusterProfileConfiguration = emptyMap();
+        if (clusterProfile != null) {
+            secretParamResolver.resolve(clusterProfile);
```

Comment:
How do we handle secret resolution failures?

We are removing the job from `jobCreationTimeMap` before secrets are resolved. Will there be any repercussions of secret resolution failures after removing it from the map.

Reply:
This is called when a job gets completed. Secret resolution failure would mean that the plugin is not notified nd hence will not be able to delete the job. Also, the job plan related information in DB is also not deleted.